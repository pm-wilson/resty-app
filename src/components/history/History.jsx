import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';
import styles from './historystyles.css';

const History = ({ history }) => {
  const historyList = history.map((historyData) => (
    <HistoryItem 
      key={historyData.key}
      url={historyData.url}
      method={historyData.method}
    />
  ));

  return (
    <div className={styles.historylist}>
      {historyList}
    </div>
  );
};

History.propTypes = {
  history: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      method: PropTypes.string.isRequired,
    })
  ),
};

export default History;
