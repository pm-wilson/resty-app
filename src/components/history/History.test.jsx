import React from 'react';
import { render, cleanup } from '@testing-library/react';
import History from './History';

describe('History component', () => {
  const data = [
    {
      key='10/22/20',
      url='futuramaapi.herokuapp.com/api/v2/characters',
      method='GET',
    {
      key='10/23/20',
      url='futuramaapi.herokuapp.com/api/quotes',
      method='GET',
  ];

  afterEach(() => cleanup());
  it('renders History', () => {
    const { asFragment } = render(<History />);
    expect(asFragment()).toMatchSnapshot();
  });
});
