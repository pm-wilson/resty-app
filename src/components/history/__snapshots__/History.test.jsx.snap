// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History 1`] = `
<DocumentFragment>
  <div
    class="historylist"
  >
    <div
      class="historyitem"
    >
      <span
        class="methodarea"
      >
        GET
      </span>
      <span
        class="urlarea"
      >
        futuramaapi.herokuapp.com/api/v2/characters
      </span>
    </div>
    <div
      class="historyitem"
    >
      <span
        class="methodarea"
      >
        GET
      </span>
      <span
        class="urlarea"
      >
        futuramaapi.herokuapp.com/api/quotes
      </span>
    </div>
  </div>
</DocumentFragment>
`;
